lib/src/model/tv/tv_controller.dart

```diff
@@ -100,6 +100,70 @@ class TvController extends _$TvController {
     state = AsyncValue.data(newState);
   }
 
+  bool canGoBack() {
```

Comment:
This code is fine and perfectly readable in my opinion, even if, I agree the `if (state.hasValue` followed by `state.requireValue` is a bit verbose and redundant.

If you're looking for alternate way, I think you could also write it like:
```dart
bool canGoBack() => state.map(
  data: (d) => d.stepCursor > 0,
  error: (_) => false,
  loading: (_) => false,
);
```

or even
```dart
bool canGoBack() => state.mapOrNull(data: (d) => d.stepCursor > 0) ?? false
```

or maybe there is even a better way? The `AsyncValue` api is powerful but also complex.

Reply:
yeah I like it better now